### PR TITLE
Add query payload encode cache

### PR DESCRIPTION
**Pull request by auxten**

### CODE REVIEW

File: types/request_type.go [hunk 1]
```diff
@@ -139,6 +140,14 @@ func (r *Request) Sign(signer *asymmetric.PrivateKey) (err error) {
 	return r.Header.Sign(signer)
 }
 
+func (r *Request) SetMarshalCache(buf []byte) {
```
Comment (covenantdb) [inline types/request_type.go:144]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Request.SetMarshalCache should have comment or be unexported

File: types/request_type.go [hunk 2]
```diff
@@ -139,6 +140,14 @@ func (r *Request) Sign(signer *asymmetric.PrivateKey) (err error) {
...
+	r._marshalCache = buf
+}
+
+func (r *Request) GetMarshalCache() (buf []byte) {
```
Comment (covenantdb) [inline types/request_type.go:149]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Request.GetMarshalCache should have comment or be unexported

**Review by leventeliu — APPROVED** [review]
LGTM.

**Review by zeqing-guo — APPROVED** [review]
Approve

**Pull request merged**